Source/WebCore/loader/cache/CachedResource.cpp

```diff
@@ -454,6 +454,9 @@ void CachedResource::redirectReceived(ResourceRequest&& request, const ResourceR
 {
     CACHEDRESOURCE_RELEASE_LOG("redirectReceived:");
 
+    if(request.url().hasFragmentIdentifier())
```

Comment:
This has a bug in the case of `localhost:10000` then loading `localhost:10000/#key=foo`.